### PR TITLE
fix(components): [table] wrong hover style in nested tables

### LINKED ISSUE #9723: [Component] [table] el-table column设置fixed  在expand插槽中嵌套表格 鼠标hover外部table行时样式错误
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.36`
- Element Plus Version: `2.2.12`
- Browser / OS: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36 Edg/105.0.1343.27`
- Build Tool: `Vite`

### Related Component

- `el-table`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtdGFibGUgOmRhdGE9XCJ0YWJsZURhdGFcIiBzdHlsZT1cIndpZHRoOiAxMDAlXCI+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBsYWJlbD1cIlwiIHR5cGU9XCJleHBhbmRcIiA6d2lkdGg9XCIxMDBcIj5cbiAgICAgIDx0ZW1wbGF0ZSAjZGVmYXVsdD1cIntyb3d9XCI+XG4gICAgICAgICAgPGVsLXRhYmxlIDpkYXRhPVwicm93LmZhbWlseVwiPlxuICAgICAgICAgICAgPGVsLXRhYmxlLWNvbHVtbiBsYWJlbD1cIk5hbWVcIiBwcm9wPVwibmFtZVwiIC8+XG4gICAgICAgICAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwiU3RhdGVcIiBwcm9wPVwic3RhdGVcIiAvPj5cbiAgICAgICAgICA8L2VsLXRhYmxlPlxuICAgICAgPC90ZW1wbGF0ZT5cbiAgICA8L2VsLXRhYmxlLWNvbHVtbj5cbiAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwiRGF0ZVwiIHByb3A9XCJkYXRlXCIgLz5cbiAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwiTmFtZVwiIHByb3A9XCJuYW1lXCIgZml4ZWQ9XCJyaWdodFwiIC8+XG4gIDwvZWwtdGFibGU+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB0YWJsZURhdGEgPSBbXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wMycsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgc3RhdGU6ICdDYWxpZm9ybmlhJyxcbiAgICBjaXR5OiAnU2FuIEZyYW5jaXNjbycsXG4gICAgYWRkcmVzczogJzM2NTAgMjFzdCBTdCwgU2FuIEZyYW5jaXNjbycsXG4gICAgemlwOiAnQ0EgOTQxMTQnLFxuICAgIGZhbWlseTogW1xuICAgICAge1xuICAgICAgICBuYW1lOiAnSmVycnknLFxuICAgICAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgICAgICBjaXR5OiAnU2FuIEZyYW5jaXNjbycsXG4gICAgICAgIGFkZHJlc3M6ICczNjUwIDIxc3QgU3QsIFNhbiBGcmFuY2lzY28nLFxuICAgICAgICB6aXA6ICdDQSA5NDExNCcsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICBuYW1lOiAnU3Bpa2UnLFxuICAgICAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgICAgICBjaXR5OiAnU2FuIEZyYW5jaXNjbycsXG4gICAgICAgIGFkZHJlc3M6ICczNjUwIDIxc3QgU3QsIFNhbiBGcmFuY2lzY28nLFxuICAgICAgICB6aXA6ICdDQSA5NDExNCcsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICBuYW1lOiAnVHlrZScsXG4gICAgICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgICAgIGNpdHk6ICdTYW4gRnJhbmNpc2NvJyxcbiAgICAgICAgYWRkcmVzczogJzM2NTAgMjFzdCBTdCwgU2FuIEZyYW5jaXNjbycsXG4gICAgICAgIHppcDogJ0NBIDk0MTE0JyxcbiAgICAgIH0sXG4gICAgXSxcbiAgfSxcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTAyJyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdTYW4gRnJhbmNpc2NvJyxcbiAgICBhZGRyZXNzOiAnMzY1MCAyMXN0IFN0LCBTYW4gRnJhbmNpc2NvJyxcbiAgICB6aXA6ICdDQSA5NDExNCcsXG4gICAgZmFtaWx5OiBbXG4gICAgICB7XG4gICAgICAgIG5hbWU6ICdKZXJyeScsXG4gICAgICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgICAgIGNpdHk6ICdTYW4gRnJhbmNpc2NvJyxcbiAgICAgICAgYWRkcmVzczogJzM2NTAgMjFzdCBTdCwgU2FuIEZyYW5jaXNjbycsXG4gICAgICAgIHppcDogJ0NBIDk0MTE0JyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIG5hbWU6ICdTcGlrZScsXG4gICAgICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgICAgIGNpdHk6ICdTYW4gRnJhbmNpc2NvJyxcbiAgICAgICAgYWRkcmVzczogJzM2NTAgMjFzdCBTdCwgU2FuIEZyYW5jaXNjbycsXG4gICAgICAgIHppcDogJ0NBIDk0MTE0JyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIG5hbWU6ICdUeWtlJyxcbiAgICAgICAgc3RhdGU6ICdDYWxpZm9ybmlhJyxcbiAgICAgICAgY2l0eTogJ1NhbiBGcmFuY2lzY28nLFxuICAgICAgICBhZGRyZXNzOiAnMzY1MCAyMXN0IFN0LCBTYW4gRnJhbmNpc2NvJyxcbiAgICAgICAgemlwOiAnQ0EgOTQxMTQnLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuICBcbl1cbjwvc2NyaXB0PlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319)

### Steps to reproduce

点击进入复现链接，鼠标移入外部table行时，hover样式错误

### What is Expected?

table hover时样式正确

### What is actually happening?

hover时选中的行错误

### Additional comments

_(empty)_

image

### PR DESCRIPTION
**Pull request by init-qy**
Please make sure these boxes are checked before submitting your PR, thank you!

closed #9723

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by init-qy** [comment]
[playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtdGFibGUgOmRhdGE9XCJ0YWJsZURhdGFcIiBzdHlsZT1cIndpZHRoOiAxMDAlXCI+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBsYWJlbD1cIlwiIHR5cGU9XCJleHBhbmRcIiA6d2lkdGg9XCIxMDBcIj5cbiAgICAgIDx0ZW1wbGF0ZSAjZGVmYXVsdD1cIntyb3d9XCI+XG4gICAgICAgICAgPGVsLXRhYmxlIDpkYXRhPVwicm93LmZhbWlseVwiPlxuICAgICAgICAgICAgPGVsLXRhYmxlLWNvbHVtbiBsYWJlbD1cIk5hbWVcIiBwcm9wPVwibmFtZVwiIC8+XG4gICAgICAgICAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwiU3RhdGVcIiBwcm9wPVwic3RhdGVcIiAvPj5cbiAgICAgICAgICA8L2VsLXRhYmxlPlxuICAgICAgPC90ZW1wbGF0ZT5cbiAgICA8L2VsLXRhYmxlLWNvbHVtbj5cbiAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwiRGF0ZVwiIHByb3A9XCJkYXRlXCIgLz5cbiAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwiTmFtZVwiIHByb3A9XCJuYW1lXCIgZml4ZWQ9XCJyaWdodFwiIC8+XG4gIDwvZWwtdGFibGU+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB0YWJsZURhdGEgPSBbXG4gIHtcbiAgICBkYXRlOiAnMjAxNi0wNS0wMycsXG4gICAgbmFtZTogJ1RvbScsXG4gICAgc3RhdGU6ICdDYWxpZm9ybmlhJyxcbiAgICBjaXR5OiAnU2FuIEZyYW5jaXNjbycsXG4gICAgYWRkcmVzczogJzM2NTAgMjFzdCBTdCwgU2FuIEZyYW5jaXNjbycsXG4gICAgemlwOiAnQ0EgOTQxMTQnLFxuICAgIGZhbWlseTogW1xuICAgICAge1xuICAgICAgICBuYW1lOiAnSmVycnknLFxuICAgICAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgICAgICBjaXR5OiAnU2FuIEZyYW5jaXNjbycsXG4gICAgICAgIGFkZHJlc3M6ICczNjUwIDIxc3QgU3QsIFNhbiBGcmFuY2lzY28nLFxuICAgICAgICB6aXA6ICdDQSA5NDExNCcsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICBuYW1lOiAnU3Bpa2UnLFxuICAgICAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgICAgICBjaXR5OiAnU2FuIEZyYW5jaXNjbycsXG4gICAgICAgIGFkZHJlc3M6ICczNjUwIDIxc3QgU3QsIFNhbiBGcmFuY2lzY28nLFxuICAgICAgICB6aXA6ICdDQSA5NDExNCcsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICBuYW1lOiAnVHlrZScsXG4gICAgICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgICAgIGNpdHk6ICdTYW4gRnJhbmNpc2NvJyxcbiAgICAgICAgYWRkcmVzczogJzM2NTAgMjFzdCBTdCwgU2FuIEZyYW5jaXNjbycsXG4gICAgICAgIHppcDogJ0NBIDk0MTE0JyxcbiAgICAgIH0sXG4gICAgXSxcbiAgfSxcbiAge1xuICAgIGRhdGU6ICcyMDE2LTA1LTAyJyxcbiAgICBuYW1lOiAnVG9tJyxcbiAgICBzdGF0ZTogJ0NhbGlmb3JuaWEnLFxuICAgIGNpdHk6ICdTYW4gRnJhbmNpc2NvJyxcbiAgICBhZGRyZXNzOiAnMzY1MCAyMXN0IFN0LCBTYW4gRnJhbmNpc2NvJyxcbiAgICB6aXA6ICdDQSA5NDExNCcsXG4gICAgZmFtaWx5OiBbXG4gICAgICB7XG4gICAgICAgIG5hbWU6ICdKZXJyeScsXG4gICAgICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgICAgIGNpdHk6ICdTYW4gRnJhbmNpc2NvJyxcbiAgICAgICAgYWRkcmVzczogJzM2NTAgMjFzdCBTdCwgU2FuIEZyYW5jaXNjbycsXG4gICAgICAgIHppcDogJ0NBIDk0MTE0JyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIG5hbWU6ICdTcGlrZScsXG4gICAgICAgIHN0YXRlOiAnQ2FsaWZvcm5pYScsXG4gICAgICAgIGNpdHk6ICdTYW4gRnJhbmNpc2NvJyxcbiAgICAgICAgYWRkcmVzczogJzM2NTAgMjFzdCBTdCwgU2FuIEZyYW5jaXNjbycsXG4gICAgICAgIHppcDogJ0NBIDk0MTE0JyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIG5hbWU6ICdUeWtlJyxcbiAgICAgICAgc3RhdGU6ICdDYWxpZm9ybmlhJyxcbiAgICAgICAgY2l0eTogJ1NhbiBGcmFuY2lzY28nLFxuICAgICAgICBhZGRyZXNzOiAnMzY1MCAyMXN0IFN0LCBTYW4gRnJhbmNpc2NvJyxcbiAgICAgICAgemlwOiAnQ0EgOTQxMTQnLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuICBcbl1cbjwvc2NyaXB0PlxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTk5MzItZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiXG4gIH1cbn0iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvcnVudGltZS1kb21AbGF0ZXN0L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3NoYXJlZEBsYXRlc3QvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy05OTMyLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICBsaW5rLmhyZWYgPSAnaHR0cHM6Ly9wcmV2aWV3LTk5MzItZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gIH0pXG59IiwiX28iOnsic2hvd0hpZGRlbiI6dHJ1ZSwic3R5bGVTb3VyY2UiOiJodHRwczovL3ByZXZpZXctOTkzMi1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyJ9fQ==)

File: packages/components/table/src/table-body/index.ts [hunk 1]
```diff
@@ -36,7 +36,14 @@ export default defineComponent({
...
       raf(() => {
-        const rows = instance?.vnode.el?.querySelectorAll(`.${ns.e('row')}`)
+        // just get first level children; fix #9723
+        const el = instance?.vnode.el as HTMLElement
+        const tempId = Date.now()
+        el?.setAttribute('tempId', String(tempId))
+        const rows = el?.querySelectorAll(
+          `[tempId="${tempId}"] > .${ns.e('row')}`
+        )
+        el?.removeAttribute('tempId')
```
Comment (tolking) [inline packages/components/table/src/table-body/index.ts:46]: Is it possible to use the class name directly without using the timestamp here?

Comment (init-qy) [reply packages/components/table/src/table-body/index.ts:46]: > Is it possible to use the class name directly without using the timestamp here?

here is the another way to get rows, which also works and passes the test

```js
        const rows = Array.from(el.children).filter((e) =>
          e?.classList.contains(`${ns.e('row')}`)
        )
```

If it's better, I'll commit changes later

Comment (tolking) [reply packages/components/table/src/table-body/index.ts:46]: looks great

**Review by tolking — APPROVED** [review]
LGTM.

**Pull request merged**